src/backend/distributed/commands/foreign_server.c

```diff
@@ -230,6 +231,14 @@ PostprocessCreateForeignServerStmt(Node *node, const char *queryString)
 
 	const bool missingOk = false;
 	ObjectAddress address = GetObjectAddressFromParseTree(node, missingOk);
+
+	DeferredErrorMessage *errMsg = DeferErrorIfHasUnsupportedDependency(&address);
+	if (errMsg != NULL)
+	{
+		RaiseDeferredError(errMsg, WARNING);
+		return NIL;
+	}
```

Comment:
Since we create commands within PreprocessCreateForeignServerStmt, we'll still try to send commands to worker nodes. Can you please move command creation codes from preprocess function to here? In general we should have EnsureDependenciesExistOnAllNodes and command creation codes in the same pre/post process functions.

Reply:
Will fix that in this pr. Also opened an issue to address this inconsistency for other DDLs too: https://github.com/citusdata/citus/issues/5839.